libc/spec/stdc.td

```diff
@@ -378,6 +378,7 @@ def StdC : StandardSpec<"stdc"> {
       [
           NamedType<"float_t">,
           NamedType<"double_t">,
+          NamedType<"float16">,
```

Comment:
We actually use `_Float16` (`Float16Type`) in the public header, but this needs to match the name of the header in llvm-libc-types for libc-hdrgen to generate the #include line.

Or should we split the `LIBC_TYPES_HAS_FLOAT16` macro and the `float16` type alias into 2 separate headers, as we said we'd only use `float16` internally?

Reply:
I think that's fine, you can add some comments to explain the discrepancy here.  Maybe we can update this part when the new header gen is implemented? @michaelrj-google

Reply:
If we move the definition of `float16` to libc/src/__support/macros/properties/types.h, then we could just rename the header that defines `LIBC_TYPES_HAS_FLOAT16` to _Float16.h and we wouldn't have this discrepancy anymore.

Reply:
I see, thanks.

Reply:
If we only define `LIBC_TYPES_HAS_FLOAT16` in the public header, does it still make it sense to have it in the llvm-libc-types directory? Each header in that directory defines a type or has a `typedef`.

Also, I just noticed that I copied the C++ `using` statement to the public header which needs to compile as C. We don't seem to have any tests that use our public headers from C, except libc/test/include/stdbit_test.c.

Reply:
Yes, we still need the macro definition in the public header, so that it can be used to guard `_Float16` functions in the generated `math.h`.

Reply:
I wasn't suggesting to move it out of the whole libc/include directory for public headers, but the llvm-libc-types subdirectory currently only stores headers that define types or `typedef`s. However, I'll leave it in llvm-libc-types for now as moving it to another subdirectory would require changes to llvm-hdrgen to generate the proper include.

Reply:
OTOH, I think we can just move the `LIBC_TYPES_HAS_FLOAT16` macro definition to `llvm-libc-macros/float16-macros.h`, directly include it in `math.h.def` and don't have to add it here.

Reply:
Oops, didn't see your comment.

Reply:
Just noticed I somehow didn't delete llvm-libc-types/float16.h.